### DISCUSSION TITLE
[Recurring Transaction] Every X days feature

### DISCUSSION THREAD

**Discussion by HenryWu01** [firefly-iii/firefly-iii#8899] [General]
I want to achieve a feature that a transaction occurs every X days.

Let's say I have a subscription that valid 31 days.  
I have subscribed it on 04/20/2024, the next payment time is 05/21/2024.  

- If I choose ``every month on the X day``, then my transaction will be paid at 05/20 which is not the case.
- I have tried to combine ``everyday`` and ``skip: 31``, but it seems like the past day is displayed also for payment?

I have set the payment start date to ``05/21/2024``, skip ``31`` days (since it renews every 31 days)
![brave_Y2zdcgSJXQ](https://github.com/firefly-iii/firefly-iii/assets/99880210/0fa8decf-f31c-42e8-ac23-f5a2377324a0)

**Comment by JC5** [comment]
Every 31 days? So it just slowly drifts away from the original subscription date?

**Reply by HenryWu01** [reply]
Yes. Many of my subscriptions they don't renew on the same day, but valid for X days (majorly 31 days).  Since the day count of month is not same (sometimes 30, sometimes 31), it will shift away from the original subscription date.

For instance (a 31 days renew subscription):
- 02/18/2024
- 03/20/2024
- 04/20/2024
- 05/21/2024

**Reply by JC5** [reply]
Oh well. It's perfectly possible to set "skip" to 31. I also see that it begins with a recurrence in the past, but it won't create transactions in the past, so that shouldn't be a problem :)

**Reply by HenryWu01** [reply]
Thank you so much for answering! But when I set recurring start date to ``21/05/2024`` and skip ``31`` days, the expected withdrawl date is incorrect. You can see on the screenshot in the first comment I have created, ``Firefly III`` think that the next recurring date is ``18/06/2024`` but actually is  ``21/06/2024``.

I have created a simple script using JavaScript and moment.js to calculate the expected payment date:
![brave_x4ojSgwnxM](https://github.com/firefly-iii/firefly-iii/assets/99880210/4579fd5f-33c6-46c4-bacb-b8cb8598c42e)
